### PR TITLE
exact match reward

### PR DESCRIPTION
**Pull request by lmeribal**

### CODE REVIEW

File: turbo_alignment/generators/chat.py [hunk 1]
```diff
@@ -111,7 +113,7 @@ def generate_from_batch_records(
                     ChatInferenceOutput(
                         input_token_ids=torch.tensor(request_output.prompt_token_ids).unsqueeze(0).cuda(),
                         input_attention_mask=records['attention_mask'][i, :].unsqueeze(0).cuda(),
-                        id=None,
+                        id=sample_ids[0],
```
Comment (alekseymalakhov11) [inline turbo_alignment/generators/chat.py:116]: why [0]?

File: turbo_alignment/generators/rm.py [hunk 2]
```diff
@@ -18,33 +18,33 @@ def __init__(self, tokenizer: PreTrainedTokenizerBase, micro_batch: int = 1, mod
...
-            #         rewards.append(-100)
-            #     else:
-            #         rewards.append(-1)
-
-            rewards = ray.get(self._model.reward_forward(records=records, index=rank % self.model_replicas))
-
-            # print('FINAL REWARDS, ', rewards)
-            # print('FINAL REAL REWARDS, ', rewards.shape, rewards)
-        return rewards  # .squeeze()
+            exact_match_goldens = read_jsonl(Path("/app/turbo-alignment/tests/fixtures/datasets/chat/chat_id_x_gold.jsonl"))
```
Comment (alekseymalakhov11) [inline turbo_alignment/generators/rm.py:27]: 🥴🥴🥴

File: turbo_alignment/generators/rm.py [hunk 3]
```diff
@@ -18,33 +18,33 @@ def __init__(self, tokenizer: PreTrainedTokenizerBase, micro_batch: int = 1, mod
...
+            id_x_pred_answer = [{'id': id, 'pred_answer': answer} for id, answer in zip(records['sample_ids'], records['sample_contents'])]
+            id_x_golden_answer = [{'id': s['id'], 'gold_answer': s['gold']} for s in exact_match_goldens]
+
+            golden_lookup = {d['id']: d['gold_answer'] for d in id_x_golden_answer}
```
Comment (alekseymalakhov11) [inline turbo_alignment/generators/rm.py:31]:
```suggestion
            golden_lookup = {s['id']:  s['gold'] for d in for s in exact_match_goldens}
```
